### DISCUSSION TITLE
v-data-table-server, show-select with return-object props does not work well.

### DISCUSSION THREAD

**Discussion by ikwabe** [vuetifyjs/vuetify#18509] [Q&A]
I am facing a problem when trying to enable the `show-select` and `return-object` props on the `v-data-table-server`. 
When selecting a check box, it returns the selected row object value, however when unselect the checkbox it adds the unchecked value to the model. The same applies to when `select all` is checked. 
Seem the issue is also happens on the documentation page [Selected values](https://vuetifyjs.com/en/components/data-tables/selection/#selected-values)
Need some help on this

**Comment by batuanec** [comment]
try to make body slot of table or item slot (chnage the default render of row), in this slot, try to iterate through items and columns and implementing the provided toggleSelect function. somethig like

```
<v-data-table-server
        v-model="selectedDataImports"
        :headers="tableHeadersComputed"
        :items-length="dataImportsTotal"
        :items="dataImportsComputed"
        show-select
        return-object
 >
<template v-slot:body="{items, columns, toggleSelect}">
  <tr v-for="item in items">
    <td v-for="column in columns">
      <v-checkbox-btn
                  v-if="column.value === 'data-table-select'"
                  :model-value="isSelected({value: item})"
                  @click="toggleSelect({value: item})"
       ></v-checkbox-btn>
      <spna v-else>
        {{ item[column.value] }}
      </span>
    </td>
  </tr>
</templete>
</v-data-table>
```

i dont know if this is correct, but seems it works... :)
hope this helo u a bit...

**Reply by ivanmrsulja** [reply]
Works like a charm, thank you!

**Reply by gihanwonderlabs** [reply]
You saved my day. thanks

**Comment by pkhuji** [comment]
This worked for me in "vuetify": "^3.5.1"
```
<v-checkbox-btn
:model-value="isSelected({ value: item.id })"
@update:modelValue="toggleSelect({ value: item.id })"
></v-checkbox-btn>
```
